### ISSUE TITLE
CVE-2024-43488: RCE vulnerability in original vscode-arduino code

### ISSUE THREAD

**Issue by ahmsec** [vscode-arduino/vscode-arduino#45]
Hello folks,

The original https://github.com/microsoft/vscode-arduino code (which this repo forked from) has a remote code execution vulnerability. That is why the extension was deprecated. Please see the link below for details. 

https://msrc.microsoft.com/update-guide/vulnerability/CVE-2024-43488

It may be worth validating whether the vulnerability was transferred to this repo.

**Comment by lersi** [comment]
Hello Ahmad,
Thanks for letting me know.
This fork is almost fully 1:1 with microsoft’s repo, the difference is only minor bug fixes and improvements.

I cannot see technical details about the vulnerability on microsoft’s page, do you know where can I find additional details so I could fix this ASAP.

**Comment by ahmsec** [comment]
@MatthewSteeves Fair enough. When I reported this vulnerability to Microsoft, they said their resolution was to deprecate the plugin. So the vulnerability may have been a contributing factor, if not the entire reason, for deprecation.

@lersi The core issue is that the plugin launches an unauthenticated localhost webserver. This webserver exposes sensitive capabilities. Anyone who can send network requests to this webserver (say a malicious webpage) can exploit the capabilities of the webserver.

To reproduce:
1. Activate the plugin
2. Run `sudo netstat -tunpl` and note the webserver's port
3. Run `curl -X POST -H 'Content-Type: application/json' --data '{"link": "https://www.example.com/"}' http://localhost:55842/api/openlink` (but first update the port).

The full RCE exploit is a bit more involved. I plan to share it at a later time. But the crux of the matter is this exposed webserver.

My suggestion would be to remove the webserver entirely, and use a different IPC mechanism between the UI and the code that invokes arduino-cli.

**Comment by MatthewSteeves** [comment]
@ahmsec  Ahh ok - interesting! Given their response, I would agree it may well have factored in. Thanks for the additional background.

**Comment by ahmsec** [comment]
Further details shared here: https://www.ahmsec.io/blog/2025/11/30/rce-in-vscode-arduino-extension/

**Comment by evoscan** [comment]
this is ridiculous,  all plugins have local servers of some sort especially strong development ones.  I wish you had not reported this.  its not exposed to the internet,  there are plenty of other ways to do RCE via a malicious webpage, so why get my favorite plugin shut down.   am annoyed at ahmsec, its hardly a flaw.

**Comment by evoscan** [comment]
looks like AI has found me something I an use in the meantime

winget install ArduinoSA.CLI --accept-package-agreements --accept-source-agreements 2>&1

Found Arduino CLI [ArduinoSA.CLI] Version 1.4.1                                                                         
This application is licensed to you by its owner.
Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
Downloading https://github.com/arduino/arduino-cli/releases/download/v1.4.1/arduino-cli_1.4.1_Windows_64bit.msi
  ██████████████████████████████  17.8 MB / 17.8 MB                                                                     
Successfully verified installer hash
Starting package install...
Successfully installed
